src/rpc/ngHandlers/AccountChannels.cpp

```diff
@@ -0,0 +1,208 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <rpc/RPC.h>
+#include <rpc/RPCHelpers.h>
+#include <rpc/ngHandlers/AccountChannels.h>
+
+namespace RPCng {
+
+void
+AccountChannelsHandler::addChannel(
+    std::vector<ChannelResponse>& jsonChannels,
+    ripple::SLE const& channelSle) const
+{
+    ChannelResponse channel;
+    channel.channelID = ripple::to_string(channelSle.key());
+    channel.account =
+        ripple::to_string(channelSle.getAccountID(ripple::sfAccount));
+    channel.accountDestination =
+        ripple::to_string(channelSle.getAccountID(ripple::sfDestination));
+    channel.amount = channelSle[ripple::sfAmount].getText();
+    channel.balance = channelSle[ripple::sfBalance].getText();
+    if (publicKeyType(channelSle[ripple::sfPublicKey]))
+    {
+        ripple::PublicKey const pk(channelSle[ripple::sfPublicKey]);
+        channel.publicKey = toBase58(ripple::TokenType::AccountPublic, pk);
+        channel.publicKeyHex = strHex(pk);
+    }
+    channel.settleDelay = channelSle[ripple::sfSettleDelay];
+    if (auto const& v = channelSle[~ripple::sfExpiration])
+        channel.expiration = *v;
+    if (auto const& v = channelSle[~ripple::sfCancelAfter])
+        channel.cancelAfter = *v;
+    if (auto const& v = channelSle[~ripple::sfSourceTag])
+        channel.sourceTag = *v;
+    if (auto const& v = channelSle[~ripple::sfDestinationTag])
+        channel.destinationTag = *v;
+
+    jsonChannels.push_back(channel);
+}
+
+AccountChannelsHandler::Result
+AccountChannelsHandler::process(AccountChannelsHandler::Input input) const
+{
+    auto range = sharedPtrBackend_->fetchLedgerRange();
```

Comment:
minor: Let's try to make everything that we can into `auto const` in this entire member function🥇